snakeoil/test/test_osutils.py

```diff
@@ -311,44 +312,127 @@ def test_nonexistent(self):
         self.assertRaises(osutils.NonExistent, osutils.FsLock,
                           pjoin(self.dir, 'missing'))
 
-    def test_locking(self):
-        path = pjoin(self.dir, 'lockfile')
+    @staticmethod
```

Comment:
Yes. It needs two separate processes to lock reliably.

Long story short, locks in *nix suck a lot more than they could reliably suck. A common bad idea that locks are per process+file rather than fd. Which means that if you got two fds for the same file, lock may be shared by them and locking ops on one of them affect the other. So two `LOCK_EX` attempt may succeed (because the file is already locked exclusively), and — the real fun thing — closing one of the fds may actually release the lock on both.